share/availability.go

```diff
@@ -3,23 +3,42 @@ package share
 import (
 	"context"
 	"errors"
+	"time"
 
 	"github.com/celestiaorg/celestia-app/pkg/da"
 )
 
 // ErrNotAvailable is returned whenever DA sampling fails.
 var ErrNotAvailable = errors.New("share: data not available")
 
+const (
+	// AvailabilityTimeout specifies timeout for DA validation during which data have to be found on the network,
+	// otherwise ErrNotAvailable is fired.
+	// TODO: https://github.com/celestiaorg/celestia-node/issues/10
+	DefaultAvailabilityTimeout = 20 * time.Minute
+
+	// SampleAmount specifies the minimum required amount of samples a light node must perform
+	// before declaring that a block is available
+	DefaultSampleAmount = 16
+
+	// DefaultWriteBatchSize defines the size of the batched header write.
+	// Headers are written in batches not to thrash the underlying Datastore with writes.
+	// TODO(@Wondertan, @renaynay): Those values must be configurable
+	// and proper defaults should be set for specific node type. (#709)
+	DefaultWriteBatchSize = 2048
+
+	// DefaultCacheAvailabilityPrefix defines the string key to be used with in
+	// the datastore as the cache availability data store key
+	DefaultCacheAvailabilityPrefix = "sampling_result"
```

Comment:
and this one. General rule is to keep only abstract top level and put everything implementation specific into subpkg